frontend/src/queries/query.ts

```diff
@@ -36,17 +36,26 @@ export async function query<N extends DataNode = DataNode>(
     refresh?: boolean
 ): Promise<N['response']> {
     if (isEventsQuery(query)) {
-        if (!query.before && !query.after) {
+        // Move HogQL filters into the "where" clause of the query. This adds support for filtering on aggregations.
+        const newQuery = { ...query }
+        const newWhere = (query.properties ?? [])
+            .filter((p) => p.type === PropertyFilterType.HogQL && p.key?.trim().length)
+            .map((p) => (p.key ?? '').trim())
+        if (newWhere.length > 0) {
+            newQuery.where = [...(query.where ?? []), ...newWhere]
+            newQuery.properties = (query.properties ?? []).filter((p) => p.type !== PropertyFilterType.HogQL)
```

Comment:
Can we do the exact same logic but in the backend? Unless I'm missing something that'd make that more complicated. Just doesn't feel like "HogQL aggregations only work in `where` and not in `properties`" is something the frontend should _have_ to know. Especially given we have a public API for this – by concentrating all the querying magic in the backend, we can keep the API dead simple.

Reply:
I tried for a long while, and realised it's likely another 4h-8h to do it properly with all the cleanup:
- I'd have to modify the `_data` var on `Filter`, as the `filter.property_groups` that `query_events_list` depends upon is just something that decodes it from within the raw "json as a string" string. 
- Even then fudging with the parsed `filter.property_groups` is annoying, because after various steps it's in the format `{ type: 'and', values: [ { type: 'and', values: [{ type: 'hogql', key; "actual_hogql(properties.bla)" }, ] }, { type: 'and', values: [{ type: 'hogql', key; "actual_hogql(properties.bla)" }, ] } ]}`
- A clean solution would need to make the `Filter` class `having`-aware, and handle the complexity of any and/or combinations thrown at it.

As filtering by HogQL aggregations doesn't make sense in the context of other insights/scenes (how would you use it in funnels/trends?), this PR was my compromise solution to unblock the frontend errors for now.

If this is rejected, my second attempt would be to inject something between `request.GET.get('properties')` and `Filter`. I'd be against pushing the `having` change into `Filter` for now, as I'm not sure if the input filter is the right object to hold details of the generated query's context, and this doesn't fit into the existing `hogql_context` that I'm [trying](https://github.com/PostHog/posthog/pull/13830) to pass in the filter. The right solution might be to move towards `hogql = convert_any_property_filter_to_hogql(filter); sql = translate_hogql(hogql)` instead 🤔... and eventually deprecating `properties` and only using `where`.

What would you do? 😅

Reply:
Ultimately this should be pushed in to the backend either by making Filter validatable and schema-ified or by replacing it.

For now IMO it's ok to keep describing the structure in the FE.

We really should describe this compromise with some tests though. So that when we come to move/replace it we don't have to relearn the problem.

it('moves the blah into the wotsit')
       expext Object A -> Object B